### PR TITLE
dart: 2.6.1 (new formula)

### PR DESCRIPTION
**Pull request by JarryShaw**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Official formula for the Dart SDK from `dart-lang/dart/dart`.

This formula is required for `dart-sass` at #47438

### CODE REVIEW

**Review by jonchang — CHANGES_REQUESTED** [review]
Formulae in Homebrew-core must build from source or produce cross-platform binaries. See https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae for details.

File: Formula/dart.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    if Hardware::CPU.is_64_bit?
+      url "https://storage.googleapis.com/dart-archive/channels/stable/release/2.6.1/sdk/dartsdk-linux-arm64-release.zip"
+      sha256 "4570933406a042fadfad1dc4b664cfcf5ab7c10ca83a0d0e9e7d5a7f362ac0b2"
+    else
+      url "https://storage.googleapis.com/dart-archive/channels/stable/release/2.6.1/sdk/dartsdk-linux-arm-release.zip"
+      sha256 "cf080fa8bdeae9b1f765ba3efab0940522506ec3077eb2751b4afd0a6448e407"
+    end
+  end
+
+  devel do
```
Comment (SMillerDev) [inline Formula/dart.rb]: Please don't use devel blocks

File: Formula/dart.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    libexec.install Dir["*"]
+    bin.install_symlink "#{libexec}/bin/dart"
+    bin.write_exec_script Dir["#{libexec}/bin/{pub,dart?*}"]
+  end
+
+  def shim_script(target)
+    <<~EOS
+      #!/usr/bin/env bash
+      exec "#{prefix}/#{target}" "$@"
+    EOS
```
Comment (SMillerDev) [inline Formula/dart.rb]: Please use the build-in envscript option.

File: Formula/dart.rb [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  end
+
+  def caveats; <<~EOS
+    Please note the path to the Dart SDK:
+      #{opt_libexec}
+  EOS
+  end
```
Comment (SMillerDev) [inline Formula/dart.rb]: This is standard for homebrew and shouldn't be a caveat.

**Comment by JarryShaw** [comment]
This is the official formula of Dart language SDK, copied directly from [`dart-lang/dart/dart`](https://github.com/dart-lang/homebrew-dart/tree/master/dart.rb).

**Comment by SMillerDev** [comment]
Taps don't need to adhere to https://docs.brew.sh/Acceptable-Formulae though, submissions to homebrew-core do.

**Comment by JarryShaw** [comment]
Understood @SMillerDev ; was just adding notes about the formula's origin :)

**Comment by JarryShaw** [comment]
And I suppose I need to use some help on this. Not even familiar with Dart itself.
Contributing this formula just because `dart-sass` at #47438 needs it.

**Comment by fxcoudert** [comment]
We're going to close this for now, it is very far from being acceptable and will require more work from  someone who knows dart.

**Pull request closed without merging**